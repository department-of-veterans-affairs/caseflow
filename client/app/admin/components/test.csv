VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN