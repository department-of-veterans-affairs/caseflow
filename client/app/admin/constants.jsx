export const ACTIONS = {
  SET_FEATURE_TOGGLES: 'SET_FEATURE_TOGGLES',
  POST_VETERAN_EXTRACT: 'POST_VETERAN_EXTRACT',
  STARTED_VETERAN_EXTRACT: 'STARTED_VETERAN_EXTRACT',
  FAILED_VETERAN_EXTRACT: 'FAILED_VETERAN_EXTRACT'
};
