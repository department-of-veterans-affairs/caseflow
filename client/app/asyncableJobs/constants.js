export const ACTIONS = {
  MANUAL_JOB_STARTED: 'MANUAL_JOB_STARTED',
  MANUAL_JOB_FAILED: 'MANUAL_JOB_FAILED',
  CLEAR_MANUAL_JOB_TYPE: 'CLEAR_MANUAL_JOB_TYPE',
};