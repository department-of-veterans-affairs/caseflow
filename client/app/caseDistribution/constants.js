import DISTRIBUTION from '../../constants/DISTRIBUTION';

const LEVERS = 'levers';

export const Constant = {
  LEVERS
};

export const sectionTitles = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]: DISTRIBUTION.ama_hearings_section_title,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]: DISTRIBUTION.ama_direct_review_section_title,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]: DISTRIBUTION.ama_evidence_submission_section_title
};

export const docketTimeGoalPriorMappings = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_hearings_docket_time_goals,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_direct_review_docket_time_goals,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_evidence_submission_docket_time_goals
};
