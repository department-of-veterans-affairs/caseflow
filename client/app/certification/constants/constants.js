// actions
//
export const UPDATE_PROGRESS_BAR = 'UPDATE_PROGRESS_BAR';
export const TOGGLE_CANCELLATION_MODAL = 'TOGGLE_CANCELLATION_MODAL';
export const CHANGE_VBMS_HEARING_DOCUMENT = 'CHANGE_VBMS_HEARING_DOCUMENT';
export const CHANGE_TYPE_OF_FORM9 = 'CHANGE_TYPE_OF_FORM9';
export const CHANGE_TYPE_OF_HEARING = 'CHANGE_TYPE_OF_HEARING';

export const CHANGE_REPRESENTATIVE_NAME = 'CHANGE_REPRESENTATIVE_NAME';
export const CHANGE_REPRESENTATIVE_TYPE = 'CHANGE_REPRESENTATIVE_TYPE';
export const CHANGE_OTHER_REPRESENTATIVE_TYPE = 'CHANGE_OTHER_REPRESENTATIVE_TYPE';

export const CHANGE_SIGN_AND_CERTIFY_FORM = 'CHANGE_SIGN_AND_CERTIFY_FORM';

export const CERTIFICATION_UPDATE_REQUEST = 'CERTIFICATION_UPDATE_REQUEST';
export const CERTIFICATION_UPDATE_FAILURE = 'CERTIFICATION_UPDATE_FAILURE';
export const CERTIFICATION_UPDATE_SUCCESS = 'CERTIFICATION_UPDATE_SUCCESS';

export const ON_CONTINUE_CLICK_FAILED = 'ON_CONTINUE_CLICK_FAILED';
export const ON_CONTINUE_CLICK_SUCCESS = 'ON_CONTINUE_CLICK_SUCCESS';

export const RESET_STATE = 'RESET_STATE';

// types of hearings
//
// TODO:
// on the backend, NO_HEARING_DESIRED
// should result in VIDEO being written to VACOLS,
// and HEARING_CANCELLED should result in a cancellation
// checkbox being checked, but the original hearing type
// should be undisturbed.
export const hearingPreferences = {
  VIDEO: 'VIDEO',
  TRAVEL_BOARD: 'TRAVEL_BOARD',
  WASHINGTON_DC: 'WASHINGTON_DC',
  HEARING_TYPE_NOT_SPECIFIED: 'HEARING_TYPE_NOT_SPECIFIED',
  NO_HEARING_DESIRED: 'NO_HEARING_DESIRED',
  HEARING_CANCELLED: 'HEARING_CANCELLED',
  NO_BOX_SELECTED: 'NO_BOX_SELECTED'
};

// form9 values
export const form9Types = {
  FORMAL_FORM9: 'FORMAL_FORM9',
  INFORMAL_FORM9: 'INFORMAL_FORM9'
};

// representation for the appellant
export const representativeTypes = {
  ATTORNEY: 'ATTORNEY',
  AGENT: 'AGENT',
  ORGANIZATION: 'ORGANIZATION',
  NONE: 'NONE',
  // TODO: should "Other be a real type"?
  OTHER: 'OTHER'
};

// was a hearing document found in VBMS?
export const vbmsHearingDocument = {
  FOUND: 'FOUND',
  NOT_FOUND: 'NOT_FOUND'
};

export const progressBarSections = {
  CHECK_DOCUMENTS: 'CHECK_DOCUMENTS',
  CONFIRM_CASE_DETAILS: 'CONFIRM_CASE_DETAILS',
  CONFIRM_HEARING: 'CONFIRM_HEARING',
  SIGN_AND_CERTIFY: 'SIGN_AND_CERTIFY'
};

export const certifyingOfficialTitles = {
  DECISION_REVIEW_OFFICER: 'DECISION_REVIEW_OFFICER',
  RATING_SPECIALIST: 'RATING_SPECIALIST',
  VETERANS_SERVICE_REPRESENTATIVE: 'VETERANS_SERVICE_REPRESENTATIVE',
  CLAIMS_ASSISTANT: 'CLAIMS_ASSISTANT',
  OTHER: 'OTHER'
};
