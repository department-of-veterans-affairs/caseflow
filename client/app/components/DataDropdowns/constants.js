import _ from 'lodash';

import HEARING_ROOMS_LIST from '../../../constants/HEARING_ROOMS_LIST.json';

export const HEARING_ROOM_OPTIONS = _.map(HEARING_ROOMS_LIST, (value, key) => ({
  label: value.label,
  value: key.toString()
}));
