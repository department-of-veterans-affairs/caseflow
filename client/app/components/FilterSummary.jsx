import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

const FilterSummary = ({ filteredByList, alternateColumnNames, clearFilteredByList }) => {
  let filterSummary = null;
  let filterListContent = [];
  const clearAllFiltersLink = <button
    onClick={() => clearFilteredByList({})}
    className="cf-btn-link cf-clear-filters-link"> Clear all filters</button>;

  // Don't show anything if there are no filters.
  if (!_.isEmpty(filteredByList)) {
    for (const filter in filteredByList) { // eslint-disable-line guard-for-in
      // This condition might be met if filters were added and then later removed,
      // as there could still bea key in the filteredByList object pointing to an empty array.
      if (filteredByList[filter].length > 0) {
        const filterContent = (<span
          key={filter}> {alternateColumnNames ? alternateColumnNames[filter] : filter} ({filteredByList[filter].length})
        </span>);

        filterListContent = filterListContent.concat(filterContent);
      }
    }

    // Don't show anything if there are no filters.
    // This may be different than the first condition because when filters are added
    // and then later removed, there may still be a key in the filteredByList object
    // pointing to an empty array.
    if (filterListContent.length > 0) {
      // Add commas between filters
      filterListContent = filterListContent.map((element, index) => {
        if (index < filterListContent.length - 1) {
          return <span key={`filter-summary-element-${index}`}>{element}, </span>;
        }

        return element;
      });

      filterSummary = (
        <div className="cf-filter-summary">
          <strong>Filtering by:</strong>
          {filterListContent}
          <span>{clearAllFiltersLink}</span>
        </div>
      );
    }
  }

  return filterSummary;
};

FilterSummary.propTypes = {
  filteredByList: PropTypes.object.isRequired,
  clearFilteredByList: PropTypes.func.isRequired,
  alternateColumnNames: PropTypes.object
};

export default FilterSummary;
