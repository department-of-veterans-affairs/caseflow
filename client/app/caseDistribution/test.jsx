/* eslint-disable react/prop-types */

import React from 'react';
import NavigationBar from '../components/NavigationBar';
import { BrowserRouter } from 'react-router-dom';
import PageRoute from '../components/PageRoute';
import AppFrame from '../components/AppFrame';
import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
import { LOGO_COLORS } from '../constants/AppConstants';
import Footer from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Footer';import CaseSearchLink from '../components/CaseSearchLink';

import CaseDistributionApp from './pages/CaseDistributionApp';

class CaseDistributionTest extends React.PureComponent {
  render() {
    const Router = this.props.router || BrowserRouter;
    const appName = 'Case Distribution';
    const tablestyle = {
      display: "block",
      overflowX: "scroll"
    };

    return (
      <Router {...this.props.routerTestProps}>
        <div>
          <NavigationBar
            wideApp
            defaultUrl={
              this.props.caseSearchHomePage || this.props.hasCaseDetailsRole ?
                '/search' :
                '/queue'
            }
            userDisplayName={this.props.userDisplayName}
            dropdownUrls={this.props.dropdownUrls}
            applicationUrls={this.props.applicationUrls}
            logoProps={{
              overlapColor: LOGO_COLORS.QUEUE.OVERLAP,
              accentColor: LOGO_COLORS.QUEUE.ACCENT,
            }}
            rightNavElement={<CaseSearchLink />}
            appName="Caseflow Admin"
          >
            <AppFrame>
              <AppSegment filledBackground>
                <div>
                  <PageRoute
                    exact
                    path={'/acd-controls/test'}
                    title="Case Distribution | Caseflow"
                    component={() => {
                      return (
                        <div>
                          <h1 id='top'>Case Distribution Levers Dashboard</h1>
                          <p>This page is for lower env use only and provides a convient place to review Case Distribution related data</p>
                          <h2>Navigation</h2>
                          <ul>
                            <li>
                              <a href="#levers"><button className="btn btn-primary">Case Distribution Levers</button></a>
                            </li>
                            <li>
                              <a href="#audit"><button className="btn btn-primary">Case Distribution Audit Levers</button></a>
                            </li>
                            <li>
                             <a href="#distribution_status"><button className="btn btn-primary">Distribution Status</button></a>
                            </li>
                          </ul>
                          <h2 id='levers'> Case Distribution Levers </h2>
                            <table id="case-table-description" className="usa-table-borderless undefined" style={tablestyle}>
                              <thead>
                                <tr>
                                  <td> ID </td>
                                  <td> Title </td>
                                  <td> Item </td>
                                  <td> Control Group </td>
                                  <td> Lever Group </td>
                                  <td> Lever Group Order </td>
                                  <td> Description </td>
                                  <td> Data Type </td>
                                  <td> Value </td>
                                  <td> Min / Max Value </td>
                                  <td> Unit </td>
                                  <td> Option Values</td>
                                  <td> Toggle Active </td>
                                  <td> Disabled In UI</td>
                                  <td> Algorithms Used </td>
                                  <td> Created At </td>
                                  <td> Updated At </td>
                                </tr>
                              </thead>
                              <tbody>
                              {this.props.acdLevers.map((lever, i) => {
                                return [
                                  <tr>
                                    <td> { lever.id } </td>
                                    <td> { lever.title } </td>
                                    <td> { lever.item } </td>
                                    <td> { lever.control_group} </td>
                                    <td> { lever.lever_group } </td>
                                    <td> { lever.lever_group_order } </td>
                                    <td> { lever.description } </td>
                                    <td> { lever.data_type } </td>
                                    <td> { lever.value } </td>
                                    <td> { lever.min_value }/{lever.max_value } </td>
                                    <td> { lever.unit } </td>
                                    <td> { lever.options?.map((option) => option.value).join(', ') } </td>
                                    <td> { lever.is_toggle_active.toString() } </td>
                                    <td> { lever.is_disabled_in_ui.toString()  } </td>
                                    <td> { lever.algorithms_used } </td>
                                    <td> { lever.created_at } </td>
                                    <td> { lever.updated_at } </td>
                                  </tr>
                                ];
                              })}
                              </tbody>
                            </table>
                          <hr/>
                          <h2 id='audit'> Case Distribution Audit Levers </h2>
                          <table>
                            <thead>
                              <tr>
                                <td> ID </td>
                                <td> Lever ID </td>
                                <td> Created At </td>
                                <td> Previous Value</td>
                                <td> Update Value </td>
                                <td> User CSS ID </td>
                                <td> User Name</td>
                                <td> Lever Title </td>
                                <td> Lever Data Type </td>
                                <td> Lever Unit </td>
                              </tr>
                            </thead>
                            <tbody>
                            {this.props.acdHistory.map((entry, i) => {
                              return [
                                  <tr>
                                    <td> { entry.id } </td>
                                    <td> { entry.case_distribution_lever_id } </td>
                                    <td> { entry.created_at } </td>
                                    <td> { entry.previous_value } </td>
                                    <td> { entry.update_value } </td>
                                    <td> { entry.user_css_id } </td>
                                    <td> { entry.user_name } </td>
                                    <td> { entry.lever_title } </td>
                                    <td> { entry.lever_data_type } </td>
                                    <td> { entry.lever_unit } </td>
                                  </tr>
                              ];
                            })}

                            </tbody>
                          </table>
                          <hr/>
                          <h2 id='distribution_status'>Distribution Status</h2>
                          <ul>
                            <li>
                              <a href="/appeals-ready-to-distribute?csv=1"><button className="btn btn-primary">Download Appeals Ready to Distribute CSV</button></a>
                            </li>
                            <li>
                              <a href="/appeals-distributed?csv=1"><button className="btn btn-primary">Download Distributed Appeals CSV</button></a>
                            </li>
                          </ul>
                          <hr/>
                          <a href="#top"><button className="btn btn-primary">Back to Top</button></a>
                        </div>
                      );
                    }}
                  />
                </div>
              </AppSegment>
            </AppFrame>
          </NavigationBar>
          <Footer
            appName={appName}
            feedbackUrl={this.props.feedbackUrl}
            buildDate={this.props.buildDate}
          />
        </div>
      </Router>

    );

  }
}

export default CaseDistributionTest;
