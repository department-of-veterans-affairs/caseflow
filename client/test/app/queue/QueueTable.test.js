import React from 'react';
import * as queueTable from 'app/queue/QueueTable';
import FilterSummary from 'app/components/FilterSummary';
import { shallow } from 'enzyme';
import { columns, createTask, summary, headerStyle, iconStyle } from 'test/data';
import Tooltip from 'app/components/Tooltip';
import { DoubleArrow } from 'app/components/RenderFunctions';
import TableFilter from 'app/components/TableFilter';
import { COLORS, LOGO_COLORS } from 'app/constants/AppConstants';
import * as glamor from 'glamor';
import classnames from 'classnames';

jest.mock('classnames');

const { default: QueueTable, HeaderRow, BodyRows, FooterRow } = queueTable;

const ColumnContent = ({ label, filter, sort, filterProps, headerStyle, iconStyle }) => (
  <span {...headerStyle}>
    <span>{label}</span>
    {sort && (
      <span {...iconStyle} aria-label={`Sort by ${label}`} role="button" tabIndex="0">
        <DoubleArrow />
      </span>
    )}
    {filter && <TableFilter {...filterProps} />}
  </span>
);

describe('QueueTable', () => {
  describe('Local Methods', () => {
    beforeEach(() => {
      jest.spyOn(queueTable, 'getColumns');
      jest.spyOn(queueTable, 'cellClasses');

      classnames.mockReturnValue('testClasses');
    });

    test('function cellClasses({ align, cellClass}) returns classes based on alignment', () => {
      // Left alignment
      const className = queueTable.cellClasses({ align: 'left' });

      expect(classnames).toHaveBeenCalledWith([queueTable.helperClasses.left, undefined]);

      // Right alignment
      queueTable.cellClasses({ align: 'right' });
      expect(classnames).toHaveBeenCalledWith([queueTable.helperClasses.right, undefined]);

      // Center alignment
      queueTable.cellClasses({ align: 'center' });
      expect(classnames).toHaveBeenCalledWith([queueTable.helperClasses.center, undefined]);

      // No alignment specified
      queueTable.cellClasses({ align: '' });
      expect(classnames).toHaveBeenCalledWith([undefined, undefined]);

      // With cellClass provieded
      queueTable.cellClasses({ align: '', cellClass: 'test' });
      expect(classnames).toHaveBeenCalledWith([undefined, 'test']);

      // Check the returned className
      expect(className).toEqual('testClasses');
    });

    test('function getColumns(props) returns value when column is not a function', () => {
      // Run the test
      const result = queueTable.getColumns({ columns: 'test' });

      // Assertions
      expect(result).toEqual('test');
    });

    test('function getColumns(props) calls column function when it is not a value', () => {
      // Run the test
      const row = {};
      const spy = jest.fn().mockReturnValue('spied');
      const result = queueTable.getColumns({ columns: spy, rowObject: row });

      // Assertions
      expect(result).toEqual('spied');
      expect(spy).toHaveBeenCalledWith(row);
    });

    test('function getCellValue(rowObject, rowId, column) returns value function when set on column', () => {});

    test('function getCellValue(rowObject, rowId, column) returns value name when set on column', () => {});

    test('function getCellValue(rowObject, rowId, column) returns empty when neither function nor name are set on column', () => {});

    test('function getCellSpan(rowObject, column) returns valueFunction results when present on column', () => {});

    test('function getCellSpan(rowObject, column) returns 1 if valueFunciton is not present on column', () => {});
  });

  describe('HeaderRow', () => {
    beforeEach(() => {
      // Mock the CSS Module
      jest.spyOn(glamor, 'css');
      jest.spyOn(glamor, 'hover');
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    test('Matches snapshot with default props', () => {
      // Setup the test
      glamor.css.mockReturnValueOnce(headerStyle);
      glamor.hover.mockReturnValue('test-class');
      const header = shallow(<HeaderRow columns={columns} rowObjects={createTask(3)} />);

      // Assertions
      expect(header).toMatchSnapshot();
      expect(header.find('tr')).toHaveLength(1);
      expect(header.find('tr').prop('role')).toEqual('row');
      expect(header.find('th')).toHaveLength(3);

      // Test each header individually
      header.find('th').map((element, i) => {
        expect(element.prop('role')).toEqual('columnheader');
        expect(element.containsMatchingElement(ColumnContent({ label: columns[i].header, headerStyle }))).toEqual(true);
      });

      // Test the CSS classes
      expect(glamor.css).toHaveBeenNthCalledWith(1, { display: 'table-row' });
      expect(glamor.css).toHaveBeenNthCalledWith(
        2,
        {
          display: 'table-cell',
          paddingLeft: '1rem',
          paddingTop: '0.3rem',
          verticalAlign: 'middle'
        },
        'test-class'
      );
      expect(glamor.hover).toHaveBeenCalledWith({ cursor: 'pointer' });
    });

    test('Can sort when not using the tasks API and getting column sort value', () => {});

    test('Can sort when the backend can sort and getting column sort value', () => {});

    test('Can filter when not using task pages API and filtering is enabled', () => {});

    test('Can filter when not using task pages API and getting filter values', () => {});

    test('Can filter when using task pages API and column has filter options', () => {});

    test('Renders tooltip when present on the column', () => {});
  });

  describe('Row', () => {});

  describe('BodyRows', () => {});

  describe('FooterRow', () => {});

  describe('Class Methods', () => {});

  describe('render()', () => {
    test('Matches snapshot with default props', () => {
      // Setup the test
      const table = shallow(
        <QueueTable columns={columns} rowObjects={createTask(3)} summary={summary} slowReRendersAreOk />
      );

      // Assertions
      expect(table).toMatchSnapshot();
      expect(table.find(HeaderRow)).toHaveLength(1);
      expect(table.find(BodyRows)).toHaveLength(1);
      expect(table.find(FooterRow)).toHaveLength(1);
      expect(table.find(FilterSummary)).toHaveLength(1);

      // Negative tests
      expect(table.find(TableFilter)).toHaveLength(0);
      expect(table.find(DoubleArrow)).toHaveLength(0);
    });
  });
});
