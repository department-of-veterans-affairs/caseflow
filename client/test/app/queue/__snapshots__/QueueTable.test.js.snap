// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueueTable HeaderRow Matches snapshot with default props 1`] = `
<thead>
  <tr
    role="row"
  >
    <th
      className="testClasses"
      key="0"
      role="columnheader"
      scope="col"
    >
      <span
        aria-label=""
        styleName="header"
      >
        <span>
          First
        </span>
      </span>
    </th>
    <th
      className="testClasses"
      key="1"
      role="columnheader"
      scope="col"
    >
      <span
        aria-label=""
        styleName="header"
      >
        <span>
          Second
        </span>
      </span>
    </th>
    <th
      className="testClasses"
      key="2"
      role="columnheader"
      scope="col"
    >
      <span
        aria-label=""
        styleName="header"
      >
        <span>
          Second
        </span>
      </span>
    </th>
  </tr>
</thead>
`;

exports[`QueueTable render() Matches snapshot with default props 1`] = `
<div
  className="cf-table-wrapper"
>
  <FilterSummary
    clearFilteredByList={[Function]}
    filteredByList={Object {}}
  />
  <table
    aria-describedby="case-table-description"
    aria-label="Appeals"
    className="usa-table-borderless undefined"
    role="table"
  >
    <caption
      className="usa-sr-only"
      id="case-table-description"
    >
      Example table summary
    </caption>
    <HeaderRow
      cachedResponses={
        Object {
          "undefined&page=1": Object {
            "task_page_count": undefined,
            "tasks": Array [
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
            ],
            "tasks_per_page": undefined,
            "total_task_count": 3,
          },
        }
      }
      columns={
        Array [
          Object {
            "header": "First",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueName": "type",
          },
        ]
      }
      currentPage={0}
      filteredByList={Object {}}
      headerClassName=""
      needsTaskRequest={true}
      rowObjects={
        Array [
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
        ]
      }
      setSortOrder={[Function]}
      sortAscending={true}
      sortColName={null}
      tasksFromApi={null}
      updateFilteredByList={[Function]}
    />
    <BodyRows
      bodyClassName=""
      cachedResponses={
        Object {
          "undefined&page=1": Object {
            "task_page_count": undefined,
            "tasks": Array [
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
              Object {
                "appeal": Object {
                  "id": 2,
                  "vacols_id": "123-0",
                  "vbms_id": "456-0",
                },
                "type": "EstablishClaim",
                "user": Object {
                  "css_id": "123-0",
                  "id": 1,
                  "station_id": "456",
                },
              },
            ],
            "tasks_per_page": undefined,
            "total_task_count": 3,
          },
        }
      }
      columns={
        Array [
          Object {
            "header": "First",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueName": "type",
          },
        ]
      }
      currentPage={0}
      filteredByList={Object {}}
      getKeyForRow={[Function]}
      needsTaskRequest={true}
      rowClassNames={[Function]}
      rowObjects={
        Array [
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
          Object {
            "appeal": Object {
              "id": 2,
              "vacols_id": "123-0",
              "vbms_id": "456-0",
            },
            "type": "EstablishClaim",
            "user": Object {
              "css_id": "123-0",
              "id": 1,
              "station_id": "456",
            },
          },
        ]
      }
      sortAscending={true}
      sortColName={null}
      tasksFromApi={null}
    />
    <FooterRow
      columns={
        Array [
          Object {
            "header": "First",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueFunction": [Function],
          },
          Object {
            "header": "Second",
            "valueName": "type",
          },
        ]
      }
      rowObjects={Array []}
    />
  </table>
</div>
`;
